Core/Src/driver_settings/dynamic_controls_choice.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * choose_dynamic_controls.c
+ *
+ *  Created on: 7 de jul de 2022
+ *      Author: JoaoPedroNunes
+ */
+
+#include "driver_settings/dynamic_controls_choice.h"
+
+#include "cmsis_os.h"
+#include "driver_settings/RTD.h"
+#include "util/CMSIS_extra/global_variables_handler.h"
+#include "util/buttons_handler.h"
+#include "util/global_instances.h"
+#include "util/global_variables.h"
+#include "util/util.h"
+
+void dynamic_controls_choice(void* argument) {
+    UNUSED(argument);
+
+    for (;;) {
+
+#ifdef DEBUG_ECU
+        extern void brkpt();
+        brkpt();
+#endif
+
+        osThreadFlagsWait(DYNAMIC_CONTROLS_CHOICE_BTN_PRESSED_FLAG, osFlagsWaitAny,
+                          osWaitForever);
+
+        bool is_RTD_active = get_individual_flag(e_ECU_control_flagsHandle, RTD_FLAG);
+
+        if (!is_RTD_active) {
+
+        	bool is_DYNAMIC_CONTROL_active = get_individual_flag(e_ECU_control_flagsHandle, DYNAMIC_CONTROL_FLAG);
+
+        	if(!is_DYNAMIC_CONTROL_active){
+        		osEventFlagsSet(e_ECU_control_flagsHandle, DYNAMIC_CONTROL_FLAG);
+        	}
+        	if(is_DYNAMIC_CONTROL_active == true){
+        		osEventFlagsClear(e_ECU_control_flagsHandle, DYNAMIC_CONTROL_FLAG);
+        	}
+        }
```

Comment:
Não precisa aninhar isso dentro de um bloco de `if`, compromete a legibilidade.
Pensa comigo: esse é o único bloco de código dessa tarefa, e que ele só vai ser executado se não estiver em RTD. Logo, se não tiver em RTD, você pode pedir pra ele “pular” essa iteração da tarefa (percebe ali em cima que o código da tarefa tá em um `for-loop` infinito). Ou seja:
```cpp
        if (!is_RTD_active)
            continue;
            
        bool is_DYNAMIC_CONTROL_active = get_individual_flag(e_ECU_control_flagsHandle, DYNAMIC_CONTROL_FLAG);
        
        ... // (resto do código)
```